exporter/sematextexporter/config.go

```diff
@@ -20,38 +21,74 @@ type Config struct {
 	// - EU
 	// - US
 	Region string `mapstructure:"region"`
+	// MetricsConfig defines the configuration specific to metrics
 	MetricsConfig `mapstructure:"metrics"`
+	// LogsConfig defines the configuration specific to logs
+	LogsConfig `mapstructure:"logs"`
 }
 
 type MetricsConfig struct {
 	// App token is the token of Sematext Monitoring App to which you want to send the metrics.
-	AppToken         string                     `mapstructure:"app_token"`
+	AppToken string `mapstructure:"app_token"`
 	// MetricsEndpoint specifies the endpoint for receiving metrics in Sematext
-	MetricsEndpoint  string                     `mapstructure:"metrics_endpoint"`
-	QueueSettings    exporterhelper.QueueConfig `mapstructure:"sending_queue"`
+	MetricsEndpoint string                     `mapstructure:"metrics_endpoint"`
+	QueueSettings   exporterhelper.QueueConfig `mapstructure:"sending_queue"`
 	// MetricsSchema indicates the metrics schema to emit to line protocol.
 	// Default: telegraf-prometheus-v2
-	MetricsSchema    string                     `mapstructure:"metrics_schema"`
+	MetricsSchema string `mapstructure:"metrics_schema"`
 	// PayloadMaxLines is the maximum number of line protocol lines to POST in a single request.
-	PayloadMaxLines  int                        `mapstructure:"payload_max_lines"`
+	PayloadMaxLines int `mapstructure:"payload_max_lines"`
 	// PayloadMaxBytes is the maximum number of line protocol bytes to POST in a single request.
-	PayloadMaxBytes  int                        `mapstructure:"payload_max_bytes"`
+	PayloadMaxBytes int `mapstructure:"payload_max_bytes"`
+}
+type LogsConfig struct {
+	// App token is the token of Sematext Monitoring App to which you want to send the logs.
+	AppToken string `mapstructure:"app_token"`
+	// LogsEndpoint specifies the endpoint for receiving logs in Sematext
+	LogsEndpoint string `mapstructure:"logs_endpoint"`
+	// LogRequests determines whether request tracking is enabled
+	LogRequests bool `mapstructure:"logs_requests"`
+	// LogMaxAge is the max number of days to retain old log files
+	LogMaxAge int `mapstructure:"logs_max_age"`
+	// LogMaxBackups is the maximum number of old log files to retain.
+	LogMaxBackups int `mapstructure:"logs_max_backups"`
+	// LogMaxSize is the maximum size in megabytes of the log file before it gets rotated
+	LogMaxSize int `mapstructure:"logs_max_size"`
+	// WriteEvents determines if events are logged
+	WriteEvents atomic.Bool `mapstructure:"write_events"`
```

Comment:
Question: Why is WriteEvents an `atomic.Bool`? Is it something that can be updated in multiple threads concurrently?

Reply:
So this was what was defined in the agent so I literally copied and pasted

Reply:
Read up on Atomic Counters in Go. It'll help you make a decision if this is really required or not